### PR TITLE
Remove global variable LOADED_PERSPECTIVES

### LINKED ISSUE #1972: Global variable for q slicer may lead to circular imports
Paul Kienzle remarked to the use of global variable here
https://github.com/SasView/sasview/blob/d3c75fe3958ad0a22b7f20ee9838e663d99696cf/src/sas/qtgui/MainWindow/GuiManager.py#L63-L64
and 
https://github.com/SasView/sasview/blob/d3c75fe3958ad0a22b7f20ee9838e663d99696cf/src/sas/qtgui/Plotting/QRangeSlider.py#L145-L147

Global variables are contraindicated, especially in this case where they lead to circular imports.
One must be able to find the graph associated with the model (otherwise you wouldn't be able to update the theory plot when parameters change in the fitting perspective). Can you use this mechanism to find the plot associated with the perspective that needs a Q range? Then you can call the Q range selector on the specific plot with a callback to adjust low/high values when the interactors are moved. In particular, the Q range slider code won't need to rummage through the perspective tabs to find the associated perspective and it won't need to know anything about the structure of the perspective to set up the associated getter/setter/input. It should also work well with a linear fit tool, which wants to use the data for selecting the range but not the associated perspective.

### PR DESCRIPTION
**Pull request by krzywon**
This PR removes the global variable LOADED_PERSPECTIVES introduced since the release of 5.0.4. The perspective is found from the parental hierarchy of the Plotter class, instead. Unit tests were updated to account for the change.

fixes #1972

### CODE REVIEW

**Review by wpotrzebowski — COMMENTED** [review]
I cannot really add more than Piotr mentioned previously. I will test functionality and therfore I've merged changes to gh actions yml files to trigger build of the branch.

**Comment by wpotrzebowski** [comment]
Testiing on OSX from installer: https://github.com/SasView/sasview/releases/tag/untagged-40d71438d507d64ee4a4. It works as expected.

File: src/sas/qtgui/Plotting/QRangeSlider.py [hunk 1]
```diff
@@ -137,14 +137,13 @@ def __init__(self, base, axes, color='black', zorder=5, x=0.5, y=0.5,
         self.draw(zorder)
         # Is the slider able to move
         self.has_move = True
-        if not perspective:
-            # The perspective this slider is actively tied to
+        try:
+            self.perspective = self.base.base.parent.manager.parent.loadedPerspectives.get(perspective, None)
```
[the remark this answers is not in the document]
Comment (krzywon) [reply src/sas/qtgui/Plotting/QRangeSlider.py]: I didn't realize this existed! Thanks for the tip. I've simplified this call now.

[the remark this answers is not in the document]
Comment (rozyczko) [reply src/sas/qtgui/Plotting/QRangeSlider.py]: I created the ObjectLibrary class originally for testing and scripting but I guess nothing prohibits us from using it in the code... it breaks some OO encapsulation rules but it's still better than traversing parent tree I think

**Review by rozyczko — APPROVED** [review]
lgtm

**Pull request merged**